### DISCUSSION TITLE
How to use the feature of external temperature scripts

### DISCUSSION THREAD

**Discussion by gigaklatch** [aaronwmorris/indi-allsky#958] [Q&A]
I do not understand how to use external scripts for temperature. 

Is there somewhere in particular where I should place the script?

What do I enter into the field for external temperature script in the config section of the web interface?

I have tried with the example scripts. Entering the absolute path to the script into the config. But either it says that the script does not exist or is not executable.

**Comment by aaronwmorris** [comment]
The script may be located anywhere, as long as it is readable and executable as the user running the indi-allsky process.

If you get the error that it is not executable, you just need to make the script executable.  `chmod 755 /path/to/script.sh`

I have updated the wiki page with a method for testing the script.
https://github.com/aaronwmorris/indi-allsky/wiki/External-Temperature-Sensor

**Reply by gigaklatch** [reply]
Great! That did the trick! The python example script worked well after making it executable. It also seemed to work adding some extra lines of codes to have it retrieve the value for the variable from elsewhere. All good.

Now, I have several temp and humidity sensors, both inside of the camera box as well as outside. So I would need to additional lines of text to the image for those. In the extra image text file field I guess I enter the path to a text file. What should be in that text file? How should it be formatted?  Do I simply make my Pi run a script that creates this text file and enters the variables equaling their their respective values into it? And then, I add into the label template field the new labels plus curly brackets containing the name of the variables in the text file and how I would like them formatted? Am I on the right track?

edit*
Scratch the last paragraph I think I now do understand how to use the extra image text file thingy.

**Comment by caphector** [comment]
I was just trying a temp script; I threw together something in Python that spits out a value but when I try to configure my script I get the error `[Errno 2] No such file or directory: '/tmp/tmpbi8z1psn.json'`.

**Reply by aaronwmorris** [reply]
The JSON data has to be written to the file referenced in the `TEMP_JSON` environment variable.  STDOUT and STDERR are not used.

**Reply by caphector** [reply]
I made a STDERR there by assuming I knew how worked. How long is the timeout on the script? My sensor is a bit flaky and can take a few seconds to return a good result.

**Reply by aaronwmorris** [reply]
I believe the current code has a hard timeout of 3 seconds.  I could make this configurable.

**Reply by caphector** [reply]
I was reading up on the sensor and it looks like it has a 0.5 hz update rate so a max of every two seconds. I sometimes get errors back from it so it can take a bit longer; how problematic would a wait for sensor data be?
